src/main/kotlin/no/nav/klage/oppgave/service/SaksbehandlerService.kt

```diff
@@ -30,4 +37,31 @@ class SaksbehandlerService(
 
     private fun getNameForIdent(it: String) =
         saksbehandlerRepository.getNamesForSaksbehandlere(setOf(it)).getOrDefault(it, "Ukjent navn")
-}
\ No newline at end of file
+
+    @Transactional
+    fun storeValgtEnhetId(ident: String, enhetId: String): EnhetMedLovligeTemaer {
+        val enhet = getEnheterMedTemaerForSaksbehandler().enheter.find { it.enhetId == enhetId }
+            ?: throw IllegalArgumentException("Saksbehandler $ident har ikke tilgang til enhet $enhetId")
```

Comment:
Lurer på om dette burde resultert i en annen type exception? Noe med rettigheter?

Reply:
Jeg vurderte det, men endte på denne. Men jeg kan godt endre, det er samme for meg.